File: 0238-product-of-array-except-self/0238-product-of-array-except-self.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* productExceptSelf(int* nums, int N, int* resultSize){
    *resultSize = N;
    int *result = malloc(sizeof(int) * N);
    for(int i = 0, p = 1; i < N; i++){
        result[i] = p;
        p *= nums[i];
    }
    for(int i = N-1, p = 1; i >- 1; i--){
        result[i] *= p;
        p *= nums[i];
    }
    return result;
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        for(int i = 0; i < nums.length; i++){
            ans[i] = 1;
        }
        for(int i = 0, p = 1; i < nums.length; i++){
            ans[i] *= p;
            p *= nums[i];
        }
        for(int i = nums.length-1, p = 1; i >= 0; i--){
            ans[i] *= p;
            p *= nums[i];
        }
        return ans;
    }
}